File: app/src/main/java/fr/free/nrw/commons/upload/UploadResult.java
package fr.free.nrw.commons.upload;

import org.wikipedia.gallery.ImageInfo;

public class UploadResult {
    private final String result;
    private final String filekey;
    private final String filename;
    private final String sessionkey;
    private final ImageInfo imageinfo;

    public UploadResult(String result, String filekey, String filename, String sessionkey, ImageInfo imageinfo) {
        this.result = result;
        this.filekey = filekey;
        this.filename = filename;
        this.sessionkey = sessionkey;
        this.imageinfo = imageinfo;
    }

    public String getResult() {
        return result;
    }

    public String getFilekey() {
        return filekey;
    }

    public String getSessionkey() {
        return sessionkey;
    }

    public ImageInfo getImageinfo() {
        return imageinfo;
    }

    public String getFilename() {
        return filename;
    }
}

File: app/src/main/java/fr/free/nrw/commons/upload/UploadResult.kt
package fr.free.nrw.commons.upload

import android.os.Parcel
import android.os.Parcelable
import org.wikipedia.gallery.ImageInfo

private const val RESULT_SUCCESS = "Success"


data class UploadResult(
    val result: String,
    val filekey: String,
    val offset: Int,
    var filename: String
) : Parcelable {
    constructor(parcel: Parcel) : this(
        result = parcel.readString() ?: "",
        filekey = parcel.readString() ?: "",
        offset = parcel.readInt() ?: 0,
        filename = parcel.readString() ?: "".trim()
    ) {
    }

    fun isSuccessful(): Boolean = result == RESULT_SUCCESS

    fun createCanonicalFileName() = "File:$filename"
    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(result)
        parcel.writeString(filekey)
        parcel.writeInt(offset)
        parcel.writeString(filename.trim())
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<UploadResult> {
        override fun createFromParcel(parcel: Parcel): UploadResult {
            return UploadResult(parcel)
        }

        override fun newArray(size: Int): Array<UploadResult?> {
            return arrayOfNulls(size)
        }
    }
}
